ingestors/kafka/src/test/scala/hydra/kafka/endpoints/BootstrapEndpointV2Spec.scala

```diff
@@ -346,6 +372,64 @@ final class BootstrapEndpointV2Spec
           }
       }.unsafeRunSync()
     }
+
+    Seq(
+      "20230101" -> "Pre cutoff date",
+      DEFAULT_LOOPHOLE_CUTOFF_DATE_DEFAULT_VALUE -> "On cutoff date"
+    ) foreach {
+      case (date, dateDescription) =>
+        val dateInstant = GenericUtils.dateStringToInstant(date)
+        val dateInMillis = Option(dateInstant.toEpochMilli)
+        Seq(
+          getTestSchema("value", createdAtDefaultValue = dateInMillis) -> Seq("createdAt"),
+          getTestSchema("value", updatedAtDefaultValue = dateInMillis) -> Seq("updatedAt"),
+          getTestSchema("value", createdAtDefaultValue = dateInMillis, updatedAtDefaultValue = dateInMillis) -> Seq("createdAt", "updatedAt")
+        ) foreach {
+          case (valueSchema, fields) =>
+            s"[$dateDescription] For existing topic, accept a request with mandatory fields having a default value - ${fields.mkString(", ")}" in {
+              testCreatedTopicProgram(topics = List("dvs.testing"), dateInstant)
+                .map { bootstrapEndpoint =>
+                  Put("/v2/topics/dvs.testing", HttpEntity(ContentTypes.`application/json`,
+                    getTopicMetadataV2Request(valueSchema))
+                  ) ~> Route.seal(bootstrapEndpoint.route) ~> check {
+                    response.status shouldBe StatusCodes.OK
+                  }
+                }
+                .unsafeRunSync()
+            }
+        }
+    }
+
+    Seq(
+      getTestSchema("value", createdAtDefaultValue = Option(Instant.now().toEpochMilli)) -> Seq("createdAt"),
+      getTestSchema("value", updatedAtDefaultValue = Option(Instant.now().toEpochMilli)) -> Seq("updatedAt"),
+      getTestSchema("value", createdAtDefaultValue = Option(Instant.now().toEpochMilli),
+        updatedAtDefaultValue = Option(Instant.now().toEpochMilli)) -> Seq("createdAt", "updatedAt")
+    ) foreach {
+      case (valueSchema, fields) =>
+        s"[Post cutoff date] reject a request with mandatory fields having a default value - ${fields.mkString(", ")}" in {
+          testCreateTopicProgram
+            .map { bootstrapEndpoint =>
+              Put("/v2/topics/dvs.testing", HttpEntity(ContentTypes.`application/json`, getTopicMetadataV2Request(valueSchema))) ~> Route.seal(
+                bootstrapEndpoint.route
+              ) ~> check {
+                val errorStringSeq = fields map { f =>
+                  s"""Required field cannot have a default value in the value schema fields: field name = $f, value schema = $valueSchema, stream type = Entity."""
+                }
+
+                response.status shouldBe StatusCodes.BadRequest
+                responseAs[String] shouldBe s"${errorStringSeq.mkString("\n")}"
+              }
+            }
+            .unsafeRunSync()
+      }
+    }
```

Comment:
Why its here and we don't have tests cases in` CreateTopicProgramSpec` where we added new logic?

Reply:
Moved the tests to CreateTopicProgramSpec.